# Impor packages

In [1]:
import pandas as pd
import numpy as np
import datetime as dt

In [2]:
import os
os.getcwd()

'C:\\Users\\qlabc\\GitHub\\user-segmentation'

# Impor data dari CSV ke DataFrame

In [3]:
df = pd.read_csv('Online Retail Data.csv', header=0)
df

order_id product_code                     product_name  quantity  \
0        493410      TEST001          This is a test product.         5   
1       C493411        21539          RETRO SPOTS BUTTER DISH        -1   
2        493412      TEST001          This is a test product.         5   
3        493413        21724  PANDA AND BUNNIES STICKER SHEET         1   
4        493413        84578   ELEPHANT TOY WITH BLUE T-SHIRT         1   
...         ...          ...                              ...       ...   
461768   539991        21618   4 WILDFLOWER BOTANICAL CANDLES         1   
461769   539991        72741            GRAND CHOCOLATECANDLE         4   
461770   539992        21470    FLOWER VINE RAFFIA FOOD COVER         1   
461771   539992        22258          FELT FARM ANIMAL RABBIT         1   
461772   539992        21155            RED RETROSPOT PEG BAG         1   

                 order_date  price  customer_id  
0       2010-01-04 09:24:00   4.50      12346.0  
1       2010-01-04 09:43:00   4.25      14590.0  
2       2010-01-04 09:53:00   4.50      12346.0  
3       2010-01-04 09:54:00   0.85          NaN  
4       2010-01-04 09:54:00   3.75          NaN  
...                     ...    ...          ...  
461768  2010-12-23 16:49:00   1.25          NaN  
461769  2010-12-23 16:49:00   1.45          NaN  
461770  2010-12-23 17:41:00   3.75          NaN  
461771  2010-12-23 17:41:00   1.25          NaN  
461772  2010-12-23 17:41:00   2.10          NaN  

[461773 rows x 7 columns]

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 461773 entries, 0 to 461772
Data columns (total 7 columns):
 #   Column        Non-Null Count   Dtype  
---  ------        --------------   -----  
 0   order_id      461773 non-null  object 
 1   product_code  461773 non-null  object 
 2   product_name  459055 non-null  object 
 3   quantity      461773 non-null  int64  
 4   order_date    461773 non-null  object 
 5   price         461773 non-null  float64
 6   customer_id   360853 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 24.7+ MB


# Data cleansing

In [5]:
df_clean = df.copy()
# membuat kolom date
df_clean['order_date'] = pd.to_datetime(df_clean['order_date'])
# menghapus semua baris tanpa customer_id
df_clean = df_clean[~df_clean['customer_id'].isna()]
# menghapus semua baris tanpa product_name
df_clean = df_clean[~df_clean['product_name'].isna()]
# membuat semua product_name berhuruf kecil
df_clean['product_name'] = df_clean['product_name'].str.lower()
# menghapus semua baris dengan product_code atau product_name test
df_clean = df_clean[(~df_clean['product_code'].str.lower().str.contains('test')) |
                    (~df_clean['product_name'].str.contains('test '))]
# membuat kolom order_status dengan nilai 'cancelled' jika order_id diawali dengan huruf 'c' dan 'delivered' jika order_id tanpa awalan huruf 'c'
df_clean['order_status'] = np.where(df_clean['order_id'].str[:1]=='C', 'cancelled', 'delivered')
# mengubah nilai quantity yang negatif menjadi positif karena nilai negatif tersebut hanya menandakan order tersebut cancelled
df_clean['quantity'] = df_clean['quantity'].abs()
# menghapus baris dengan price bernilai negatif
df_clean = df_clean[df_clean['price']>0]
# membuat nilai amount, yaitu perkalian antara quantity dan price
df_clean['amount'] = df_clean['quantity'] * df_clean['price']
# mengganti product_name dari product_code yang memiliki beberapa product_name dengan salah satu product_name-nya yang paling sering muncul
most_freq_product_name = df_clean.groupby(['product_code','product_name'], as_index=False).agg(order_cnt=('order_id','nunique')).sort_values(['product_code','order_cnt'], ascending=[True,False])
most_freq_product_name['rank'] = most_freq_product_name.groupby('product_code')['order_cnt'].rank(method='first', ascending=False)
most_freq_product_name = most_freq_product_name[most_freq_product_name['rank']==1].drop(columns=['order_cnt','rank'])
df_clean = df_clean.merge(most_freq_product_name.rename(columns={'product_name':'most_freq_product_name'}), how='left', on='product_code')
df_clean['product_name'] = df_clean['most_freq_product_name']
df_clean = df_clean.drop(columns='most_freq_product_name')
# mengkonversi customer_id menjadi string
df_clean['customer_id'] = df_clean['customer_id'].astype(str)
# menghapus outlier
from scipy import stats
df_clean = df_clean[(np.abs(stats.zscore(df_clean[['quantity','amount']]))<3).all(axis=1)]
df_clean = df_clean.reset_index(drop=True)
df_clean

order_id product_code                       product_name  quantity  \
0       C493411        21539          red retrospot butter dish         1   
1        493414        21844                  red retrospot mug        36   
2        493414        21533          retro spot large milk jug        12   
3        493414        37508    new england ceramic cake server         2   
4        493414       35001G               hand open shape gold         2   
...         ...          ...                                ...       ...   
358464   539988        84380  set of 3 butterfly cookie cutters         1   
358465   539988       84849D              hot baths soap holder         1   
358466   539988       84849B             fairy soap soap holder         1   
358467   539988        22854        cream sweetheart egg holder         2   
358468   539988       47559B                tea time oven glove         2   

                order_date  price customer_id order_status  amount  
0      2010-01-04 09:43:00   4.25     14590.0    cancelled    4.25  
1      2010-01-04 10:28:00   2.55     14590.0    delivered   91.80  
2      2010-01-04 10:28:00   4.25     14590.0    delivered   51.00  
3      2010-01-04 10:28:00   2.55     14590.0    delivered    5.10  
4      2010-01-04 10:28:00   4.25     14590.0    delivered    8.50  
...                    ...    ...         ...          ...     ...  
358464 2010-12-23 16:06:00   1.25     18116.0    delivered    1.25  
358465 2010-12-23 16:06:00   1.69     18116.0    delivered    1.69  
358466 2010-12-23 16:06:00   1.69     18116.0    delivered    1.69  
358467 2010-12-23 16:06:00   4.95     18116.0    delivered    9.90  
358468 2010-12-23 16:06:00   1.25     18116.0    delivered    2.50  

[358469 rows x 9 columns]

In [6]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 358469 entries, 0 to 358468
Data columns (total 9 columns):
 #   Column        Non-Null Count   Dtype         
---  ------        --------------   -----         
 0   order_id      358469 non-null  object        
 1   product_code  358469 non-null  object        
 2   product_name  358469 non-null  object        
 3   quantity      358469 non-null  int64         
 4   order_date    358469 non-null  datetime64[ns]
 5   price         358469 non-null  float64       
 6   customer_id   358469 non-null  object        
 7   order_status  358469 non-null  object        
 8   amount        358469 non-null  float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(5)
memory usage: 24.6+ MB


# Membuat RFM segmentation

## Agregat data transaksi ke bentuk summary total transaksi (order), total nilai order (order value), tanggal order terakhir dari setiap pengguna

In [7]:
df_user = df_clean.groupby('customer_id', as_index=False).agg(order_cnt=('order_id','nunique'),max_order_date=('order_date','max'),total_order_value=('amount','sum'))
df_user

customer_id  order_cnt      max_order_date  total_order_value
0        12346.0          5 2010-10-04 09:54:00             602.40
1        12608.0          1 2010-10-31 10:49:00             415.79
2        12745.0          2 2010-08-10 10:14:00             723.85
3        12746.0          2 2010-06-30 08:19:00             266.35
4        12747.0         19 2010-12-13 10:41:00            4094.79
...          ...        ...                 ...                ...
3884     18283.0          6 2010-11-22 15:30:00             641.77
3885     18284.0          2 2010-10-06 12:31:00             486.68
3886     18285.0          1 2010-02-17 10:24:00             427.00
3887     18286.0          2 2010-08-20 11:57:00             941.48
3888     18287.0          4 2010-11-22 11:51:00            2345.71

[3889 rows x 4 columns]

## Buat kolom jumlah hari sejak order terakhir

In [8]:
today = df_clean['order_date'].max()
df_user['day_since_last_order'] = (today - df_user['max_order_date']).dt.days
df_user

customer_id  order_cnt      max_order_date  total_order_value  \
0        12346.0          5 2010-10-04 09:54:00             602.40   
1        12608.0          1 2010-10-31 10:49:00             415.79   
2        12745.0          2 2010-08-10 10:14:00             723.85   
3        12746.0          2 2010-06-30 08:19:00             266.35   
4        12747.0         19 2010-12-13 10:41:00            4094.79   
...          ...        ...                 ...                ...   
3884     18283.0          6 2010-11-22 15:30:00             641.77   
3885     18284.0          2 2010-10-06 12:31:00             486.68   
3886     18285.0          1 2010-02-17 10:24:00             427.00   
3887     18286.0          2 2010-08-20 11:57:00             941.48   
3888     18287.0          4 2010-11-22 11:51:00            2345.71   

      day_since_last_order  
0                       80  
1                       53  
2                      135  
3                      176  
4                       10  
...                    ...  
3884                    31  
3885                    78  
3886                   309  
3887                   125  
3888                    31  

[3889 rows x 5 columns]

In [9]:
today

Timestamp('2010-12-23 16:06:00')

In [10]:
df_user.describe()

order_cnt                 max_order_date  total_order_value  \
count  3889.000000                           3889        3889.000000   
mean      5.128568  2010-09-23 18:15:51.267678208        1544.623084   
min       1.000000            2010-01-05 12:43:00           1.250000   
25%       1.000000            2010-08-19 12:30:00         296.360000   
50%       3.000000            2010-10-26 18:45:00         648.200000   
75%       6.000000            2010-11-28 14:54:00        1585.940000   
max     163.000000            2010-12-23 16:06:00       71970.390000   
std       8.499330                            NaN        3434.816315   

       day_since_last_order  
count           3889.000000  
mean              90.651581  
min                0.000000  
25%               25.000000  
50%               57.000000  
75%              126.000000  
max              352.000000  
std               88.883201

## Buat binning dari jumlah hari sejak order terakhir yang terdiri dari 5 bins dengan batas-batasnya merupakan min, P20, P40, P60, P80, max dan beri label 1 sampai 5 dari bin tertinggi ke terendah sebagai skor recency

In [11]:
df_user['recency_score'] = pd.cut(df_user['day_since_last_order'],
                                  bins=[df_user['day_since_last_order'].min(),
                                        np.percentile(df_user['day_since_last_order'], 20),
                                        np.percentile(df_user['day_since_last_order'], 40),
                                        np.percentile(df_user['day_since_last_order'], 60),
                                        np.percentile(df_user['day_since_last_order'], 80),
                                        df_user['day_since_last_order'].max()],
                                  labels=[5, 4, 3, 2, 1],
                                  include_lowest=True).astype(int)
df_user

customer_id  order_cnt      max_order_date  total_order_value  \
0        12346.0          5 2010-10-04 09:54:00             602.40   
1        12608.0          1 2010-10-31 10:49:00             415.79   
2        12745.0          2 2010-08-10 10:14:00             723.85   
3        12746.0          2 2010-06-30 08:19:00             266.35   
4        12747.0         19 2010-12-13 10:41:00            4094.79   
...          ...        ...                 ...                ...   
3884     18283.0          6 2010-11-22 15:30:00             641.77   
3885     18284.0          2 2010-10-06 12:31:00             486.68   
3886     18285.0          1 2010-02-17 10:24:00             427.00   
3887     18286.0          2 2010-08-20 11:57:00             941.48   
3888     18287.0          4 2010-11-22 11:51:00            2345.71   

      day_since_last_order  recency_score  
0                       80              2  
1                       53              3  
2                      135              2  
3                      176              1  
4                       10              5  
...                    ...            ...  
3884                    31              4  
3885                    78              2  
3886                   309              1  
3887                   125              2  
3888                    31              4  

[3889 rows x 6 columns]

In [12]:
df_user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3889 entries, 0 to 3888
Data columns (total 6 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   customer_id           3889 non-null   object        
 1   order_cnt             3889 non-null   int64         
 2   max_order_date        3889 non-null   datetime64[ns]
 3   total_order_value     3889 non-null   float64       
 4   day_since_last_order  3889 non-null   int64         
 5   recency_score         3889 non-null   int32         
dtypes: datetime64[ns](1), float64(1), int32(1), int64(2), object(1)
memory usage: 167.2+ KB


## Buat binning dari total transaksi (order) yang terdiri dari 5 bins dengan batas-batasnya merupakan min, P20, P40, P60, P80, max dan beri label 1 sampai 5 dari bin terendah ke tertinggi sebagai skor frequency

In [13]:
df_user['frequency_score'] = pd.cut(df_user['order_cnt'],
                                    # membuat batas bawah bins menjadi 0
                                    bins=[0,
                                          np.percentile(df_user['order_cnt'], 20),
                                          np.percentile(df_user['order_cnt'], 40),
                                          np.percentile(df_user['order_cnt'], 60),
                                          np.percentile(df_user['order_cnt'], 80),
                                          df_user['order_cnt'].max()],
                                    labels=[1, 2, 3, 4, 5],
                                    include_lowest=True).astype(int)
df_user

customer_id  order_cnt      max_order_date  total_order_value  \
0        12346.0          5 2010-10-04 09:54:00             602.40   
1        12608.0          1 2010-10-31 10:49:00             415.79   
2        12745.0          2 2010-08-10 10:14:00             723.85   
3        12746.0          2 2010-06-30 08:19:00             266.35   
4        12747.0         19 2010-12-13 10:41:00            4094.79   
...          ...        ...                 ...                ...   
3884     18283.0          6 2010-11-22 15:30:00             641.77   
3885     18284.0          2 2010-10-06 12:31:00             486.68   
3886     18285.0          1 2010-02-17 10:24:00             427.00   
3887     18286.0          2 2010-08-20 11:57:00             941.48   
3888     18287.0          4 2010-11-22 11:51:00            2345.71   

      day_since_last_order  recency_score  frequency_score  
0                       80              2                4  
1                       53              3                1  
2                      135              2                2  
3                      176              1                2  
4                       10              5                5  
...                    ...            ...              ...  
3884                    31              4                4  
3885                    78              2                2  
3886                   309              1                1  
3887                   125              2                2  
3888                    31              4                3  

[3889 rows x 7 columns]

In [14]:
df_user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3889 entries, 0 to 3888
Data columns (total 7 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   customer_id           3889 non-null   object        
 1   order_cnt             3889 non-null   int64         
 2   max_order_date        3889 non-null   datetime64[ns]
 3   total_order_value     3889 non-null   float64       
 4   day_since_last_order  3889 non-null   int64         
 5   recency_score         3889 non-null   int32         
 6   frequency_score       3889 non-null   int32         
dtypes: datetime64[ns](1), float64(1), int32(2), int64(2), object(1)
memory usage: 182.4+ KB


## Buat binning dari total nilai order (order value) yang terdiri dari 5 bins dengan batas-batasnya merupakan min, P20, P40, P60, P80, max dan beri label 1 sampai 5 dari bin terendah ke tertinggi sebagai skor monetary

In [15]:
df_user['monetary_score'] = pd.cut(df_user['total_order_value'],
                                   bins=[df_user['total_order_value'].min(),
                                         np.percentile(df_user['total_order_value'], 20),
                                         np.percentile(df_user['total_order_value'], 40),
                                         np.percentile(df_user['total_order_value'], 60),
                                         np.percentile(df_user['total_order_value'], 80),
                                         df_user['total_order_value'].max()],
                                   labels=[1, 2, 3, 4, 5],
                                   include_lowest=True).astype(int)
df_user

customer_id  order_cnt      max_order_date  total_order_value  \
0        12346.0          5 2010-10-04 09:54:00             602.40   
1        12608.0          1 2010-10-31 10:49:00             415.79   
2        12745.0          2 2010-08-10 10:14:00             723.85   
3        12746.0          2 2010-06-30 08:19:00             266.35   
4        12747.0         19 2010-12-13 10:41:00            4094.79   
...          ...        ...                 ...                ...   
3884     18283.0          6 2010-11-22 15:30:00             641.77   
3885     18284.0          2 2010-10-06 12:31:00             486.68   
3886     18285.0          1 2010-02-17 10:24:00             427.00   
3887     18286.0          2 2010-08-20 11:57:00             941.48   
3888     18287.0          4 2010-11-22 11:51:00            2345.71   

      day_since_last_order  recency_score  frequency_score  monetary_score  
0                       80              2                4               3  
1                       53              3                1               2  
2                      135              2                2               3  
3                      176              1                2               2  
4                       10              5                5               5  
...                    ...            ...              ...             ...  
3884                    31              4                4               3  
3885                    78              2                2               3  
3886                   309              1                1               2  
3887                   125              2                2               4  
3888                    31              4                3               5  

[3889 rows x 8 columns]

In [16]:
df_user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3889 entries, 0 to 3888
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   customer_id           3889 non-null   object        
 1   order_cnt             3889 non-null   int64         
 2   max_order_date        3889 non-null   datetime64[ns]
 3   total_order_value     3889 non-null   float64       
 4   day_since_last_order  3889 non-null   int64         
 5   recency_score         3889 non-null   int32         
 6   frequency_score       3889 non-null   int32         
 7   monetary_score        3889 non-null   int32         
dtypes: datetime64[ns](1), float64(1), int32(3), int64(2), object(1)
memory usage: 197.6+ KB


## Buat kolom nama segmen berdasarkan skor recency dan frequency

In [17]:
df_user['segment'] = np.select(
    [(df_user['recency_score']==5) & (df_user['frequency_score']>=4),
     (df_user['recency_score'].between(3, 4)) & (df_user['frequency_score']>=4),
     (df_user['recency_score']>=4) & (df_user['frequency_score'].between(2, 3)),
     (df_user['recency_score']<=2) & (df_user['frequency_score']==5),
     (df_user['recency_score']==3) & (df_user['frequency_score']==3),
     (df_user['recency_score']==5) & (df_user['frequency_score']==1),
     (df_user['recency_score']==4) & (df_user['frequency_score']==1),
     (df_user['recency_score']<=2) & (df_user['frequency_score'].between(3, 4)),
     (df_user['recency_score']==3) & (df_user['frequency_score']<=2),
     (df_user['recency_score']<=2) & (df_user['frequency_score']<=2)],
    ['01-Champion', '02-Loyal Customers', '03-Potential Loyalists', "04-Can't Lose Them", '05-Need Attention',
     '06-New Customers', '07-Promising', '08-At Risk', '09-About to Sleep', '10-Hibernating']
)
df_user

customer_id  order_cnt      max_order_date  total_order_value  \
0        12346.0          5 2010-10-04 09:54:00             602.40   
1        12608.0          1 2010-10-31 10:49:00             415.79   
2        12745.0          2 2010-08-10 10:14:00             723.85   
3        12746.0          2 2010-06-30 08:19:00             266.35   
4        12747.0         19 2010-12-13 10:41:00            4094.79   
...          ...        ...                 ...                ...   
3884     18283.0          6 2010-11-22 15:30:00             641.77   
3885     18284.0          2 2010-10-06 12:31:00             486.68   
3886     18285.0          1 2010-02-17 10:24:00             427.00   
3887     18286.0          2 2010-08-20 11:57:00             941.48   
3888     18287.0          4 2010-11-22 11:51:00            2345.71   

      day_since_last_order  recency_score  frequency_score  monetary_score  \
0                       80              2                4               3   
1                       53              3                1               2   
2                      135              2                2               3   
3                      176              1                2               2   
4                       10              5                5               5   
...                    ...            ...              ...             ...   
3884                    31              4                4               3   
3885                    78              2                2               3   
3886                   309              1                1               2   
3887                   125              2                2               4   
3888                    31              4                3               5   

                     segment  
0                 08-At Risk  
1          09-About to Sleep  
2             10-Hibernating  
3             10-Hibernating  
4                01-Champion  
...                      ...  
3884      02-Loyal Customers  
3885          10-Hibernating  
3886          10-Hibernating  
3887          10-Hibernating  
3888  03-Potential Loyalists  

[3889 rows x 9 columns]

In [18]:
df_user.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3889 entries, 0 to 3888
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype         
---  ------                --------------  -----         
 0   customer_id           3889 non-null   object        
 1   order_cnt             3889 non-null   int64         
 2   max_order_date        3889 non-null   datetime64[ns]
 3   total_order_value     3889 non-null   float64       
 4   day_since_last_order  3889 non-null   int64         
 5   recency_score         3889 non-null   int32         
 6   frequency_score       3889 non-null   int32         
 7   monetary_score        3889 non-null   int32         
 8   segment               3889 non-null   object        
dtypes: datetime64[ns](1), float64(1), int32(3), int64(2), object(2)
memory usage: 228.0+ KB


## Tampilkan summary dari RFM segmentation (poin 8) berupa banyaknya pengguna, rata-rata dan median dari total order, total order value, dan jumlah hari sejak order terakhir

In [20]:
summary = pd.pivot_table(df_user, index='segment',
               values=['customer_id','day_since_last_order','order_cnt','total_order_value'],
               aggfunc={'customer_id': pd.Series.nunique,
                        'day_since_last_order': [np.mean, np.median],
                        'order_cnt': [np.mean, np.median],
                        'total_order_value': [np.mean, np.median]})
summary['pct_unique'] = (summary['customer_id'] / summary['customer_id'].sum() * 100).round(1)
summary['pct_unique'] = summary['pct_unique'].astype(str) + '%'
summary

C:\Users\qlabc\AppData\Local\Temp\ipykernel_15200\1708354394.py:1: FutureWarning: The provided callable <function mean at 0x000002906264AC00> is currently using SeriesGroupBy.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "mean" instead.
  summary = pd.pivot_table(df_user, index='segment',
C:\Users\qlabc\AppData\Local\Temp\ipykernel_15200\1708354394.py:1: FutureWarning: The provided callable <function median at 0x0000029062775E40> is currently using SeriesGroupBy.median. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "median" instead.
  summary = pd.pivot_table(df_user, index='segment',


customer_id day_since_last_order         order_cnt  \
                           nunique                 mean median       mean   
segment                                                                     
01-Champion                    553            10.533454    9.0  15.432188   
02-Loyal Customers             549            41.200364   37.0   8.744991   
03-Potential Loyalists         514            23.083658   24.0   2.830739   
04-Can't Lose Them              62           123.274194  113.0  11.467742   
05-Need Attention              184            58.505435   59.0   3.402174   
06-New Customers                50            14.000000   16.0   1.000000   
07-Promising                   133            31.954887   32.0   1.000000   
08-At Risk                     418           141.531100  120.0   4.126794   
09-About to Sleep              370            58.175676   58.0   1.416216   
10-Hibernating                1056           197.151515  199.0   1.312500   

                              total_order_value           pct_unique  
                       median              mean    median             
segment                                                               
01-Champion              10.0       4989.208761  2773.910      14.2%  
02-Loyal Customers        7.0       2618.121117  1937.050      14.1%  
03-Potential Loyalists    3.0        766.076265   621.005      13.2%  
04-Can't Lose Them       10.0       2851.737258  2268.405       1.6%  
05-Need Attention         3.0       1004.317071   826.370       4.7%  
06-New Customers          1.0        244.689000   193.675       1.3%  
07-Promising              1.0        288.694135   239.460       3.4%  
08-At Risk                4.0       1141.224835   866.320      10.7%  
09-About to Sleep         1.0        448.176081   336.735       9.5%  
10-Hibernating            1.0        342.618450   256.900      27.2%

Dari data pengguna, terlihat bahwa segmen dengan jumlah pengguna terbanyak adalah Hibernating (27.2%), Champion (14.2%), dan Loyal Customers (14.1%).

Untuk meningkatkan konversi dari segmen Loyal Customers (14.1%), dapat dilakukan program khusus dengan fokus pada meningkatkan urgensi bertransaksi sehingga mereka dapat naik ke segmen Champion.

Demikian pula, untuk segmen Potential Loyalists (13.2%), program khusus yang mengedepankan peningkatan jumlah transaksi dapat membantu mereka naik ke segmen Champion.